ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -194,6 +197,19 @@ void compact(CompactionDetails details, byte[] bundleReadBuffer) throws IOExcept
     resumeCompaction(bundleReadBuffer);
   }
 
+  /**
+   * Converts the set of {@link Container}s to the set of accountId & ContainerId pairs for easy detection.
+   */
+  private void getAllContainersByStatus(Container.ContainerStatus containerStatus) {
+    Set<Pair<Short, Short>> selectedContainerPairSet = new HashSet<>();
+    if (accountService != null) {
+      for (Container container : accountService.getContainersByStatus(containerStatus)) {
+        selectedContainerPairSet.add(new Pair<>(container.getParentAccountId(), container.getId()));
+      }
+    }
+    selectedContainers.addAll(selectedContainerPairSet);
```

Comment:
should clear `selectedcContainers` before adding item to it since we are doing for every compaction.

Reply:
Updated.